static/js/compose.js

```diff
@@ -1044,6 +1044,27 @@ export function warn_if_private_stream_is_linked(linked_stream) {
     warning_area.show();
 }
 
+export function warn_if_mentioned_private_one_on_one(mentioned) {
+    if (compose_state.get_message_type() !== "private") {
+        return;
+    }
+
+    if (compose_pm_pill.get_user_ids().length === 1) {
+        // display warning
+        const error_msg =
+            "Mentioning " +
+            mentioned.full_name +
+            " will highlight this message. Did you mean to use a silent mention (@_" +
+            mentioned.full_name +
+            ") instead?";
```

Comment:
Can you do this with an HTML template instead?  That will naturally fix the HTML escaping bug you probably have with this logic.